### PR TITLE
Tesla : improve lifecycle handling

### PR DESCRIPTION
**Pull request by kgoderis**
Fixes https://github.com/eclipse/smarthome/issues/2945

Signed-off-by: Karel Goderis <karel.goderis@me.com>

### CODE REVIEW

File: addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java [hunk 1]
```diff
@@ -137,35 +137,42 @@ public void initialize() {
 
         logger.trace("Initializing the Tesla handler for {}", getThing().getUID());
 
+        updateStatus(ThingStatus.OFFLINE);
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java]: Wouldn't be UNKNOWN the better option?

Comment (kgoderis) [inline addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java]: @kaikreuzer Maybe, maybe not. As this point in the code, from the binding's point of view, it is certain that the Thing is OFFLINE as none of the threads to make the connection to the car have actually started.

Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java]: Well, if you are still initializing your connection for the first time, you better don't set any status. If you know that within a few seconds you can set it to ONLINE or OFFLINE, better wait for it.

**Comment by kgoderis** [comment]
@kaikreuzer Included your suggestion

**Pull request merged**